### PR TITLE
Improve error message when parsing invalid spec

### PR DESCRIPTION
**Pull request by mdsgabriel**
*Issue #, if available:*

*Description of changes:*
The previous code was ignoring the error returned when calling
yaml.Unmarshal() to parse the yaml block representing a cluster
definition, giving users the following message:

./eksctl anywhere create -f test.yaml
Error: unable to find kind Cluster in file

With the new code users will see the following:

./eksctl anywhere create -f test.yaml
Error: unable to parse file test.yaml ...  line 6: did not find expected key

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/368#" title="Author self-approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/api/v1alpha1/cluster.go
Comment (vivek-koppuru) [inline pkg/api/v1alpha1/cluster.go:223]:
```suggestion
	return fmt.Errorf("cluster spec file %s is invalid or does not contain kind %s", fileName, clusterConfig.ExpectedKind())
```

**Review by vivek-koppuru — APPROVED** [review]
Nice, this is going to be a lot better!

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/368#" title="Author self-approved">mdsgabriel</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/368#pullrequestreview-769451631" title="Approved">vivek-koppuru</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mdsgabriel,vivek-koppuru]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/368#" title="Author self-approved">mdsgabriel</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/368#pullrequestreview-770429705" title="LGTM">TerryHowe</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/368#pullrequestreview-769451631" title="Approved">vivek-koppuru</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [TerryHowe,mdsgabriel,vivek-koppuru]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jiayiwang7 — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/368#pullrequestreview-770464703" title="Approved">jiayiwang7</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/368#" title="Author self-approved">mdsgabriel</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/368#pullrequestreview-770429705" title="LGTM">TerryHowe</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/368#pullrequestreview-769451631" title="Approved">vivek-koppuru</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [TerryHowe,jiayiwang7,mdsgabriel,vivek-koppuru]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
